src/Stratis.Bitcoin.Features.RPC/RpcSettings.cs

```diff
@@ -31,58 +31,91 @@ public class RpcSettings
         /// <summary>TCP port for RPC interface.</summary>
         public int RPCPort { get; set; }
 
-        /// <summary>Default bindings from config.</summary>
-        public List<IPEndPoint> DefaultBindings { get; set; }
-
         /// <summary>List of network endpoints that the node will listen and provide RPC on.</summary>
         public List<IPEndPoint> Bind { get; set; }
 
         /// <summary>List of IP addresses that are allowed to connect to RPC interfaces.</summary>
         public List<IPAddressBlock> AllowIp { get; set; }
 
+        // TODO: Refactor static PrintHelp method.
+        private static RpcSettings settingsHelp;
+
+        public List<IPEndPoint> DefaultBinding() => DefaultBinding(this.AllowIp.Any(), this.RPCPort);
+
+        private static List<IPEndPoint> DefaultBinding(bool haveAllowIps, int rpcPort)
+        {
+            var defaultBinding = new List<IPEndPoint>();
+
+            if (haveAllowIps)
+            {
+                defaultBinding.Add(new IPEndPoint(IPAddress.Parse("::"), rpcPort));
+                defaultBinding.Add(new IPEndPoint(IPAddress.Parse("0.0.0.0"), rpcPort));
+            }
+            else
+            {
+                defaultBinding.Add(new IPEndPoint(IPAddress.Parse("::1"), rpcPort));
+                defaultBinding.Add(new IPEndPoint(IPAddress.Parse("127.0.0.1"), rpcPort));
+            }
+
+            return defaultBinding;
+        }
+
         /// <summary>
         /// Initializes an instance of the object from the node configuration.
         /// </summary>
         /// <param name="nodeSettings">The node configuration.</param>
-        public RpcSettings(NodeSettings nodeSettings)
+        /// <param name="defaultArgs">Allows overriding the default values before config and command line are applied.</param>
+        public RpcSettings(NodeSettings nodeSettings, Action<RpcSettings> defaultArgs = null)
         {
             Guard.NotNull(nodeSettings, nameof(nodeSettings));
 
             this.logger = nodeSettings.LoggerFactory.CreateLogger(typeof(RpcSettings).FullName);
 
+            this.RPCPort = nodeSettings.Network.DefaultRPCPort;
             this.Bind = new List<IPEndPoint>();
-            this.DefaultBindings = new List<IPEndPoint>();
             this.AllowIp = new List<IPAddressBlock>();
+            this.Server = false;
+            this.RpcUser = null;
+            this.RpcPassword = null;
 
-            // Get values from config
+            defaultArgs?.Invoke(this);
```

Comment:
I am wondering do we really need to go back to this practice if we allow to overwrite via args?

Reply:
This is for the application to establish its own network-specific defaults without having to rely on settings being overridden via the command line. The network-specific defaults established in this manner can then be overridden by the command line and configuration file.

Reply:
Network specific defaults sound snice but when will we use it?

Reply:
We can have PrintHelp show different default settings for Bitcoin and Stratis even if some of those have to be derived by the application.